### PR TITLE
Add wedding fields

### LINKED ISSUE #108: Add fields to wedding
Compulsory fields: 2 partners
Optional Fields: location and date

Cannot unassign the partners from the wedding.
When adding the partners, also must do assign-wedding command for them.

### LINKED ISSUE #119: Remove use of RawTypes & Unchecked typecastings

### PR DESCRIPTION
**Pull request by riccoljy**
Closes #108 and fixes #119 

Add the following fields:
`Person` partner1
`Person` partner2
`ArrayList<Person>` guestList
`Address` address
`String` date

UI display of all above except guestList complete

Change `storage` to adapt to these new attributes (`JsonSerializableAddressBook` & `JsonAdaptedWedding`)

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/wedding/EditWeddingCommand.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        model.updateFilteredWeddingList(PREDICATE_SHOW_ALL_WEDDINGS);
+        return new CommandResult(String.format(MESSAGE_EDIT_WEDDING_SUCCESS, Messages.format(editedWedding)));
+    }
+
+    /**
+     * Creates and returns a {@code Wedding} with the details of {@code weddingToEdit}
+     * edited with {@code editWeddingDescriptor}.
+     */
+    private static Wedding createEditedWedding(Wedding weddingToEdit, EditWeddingDescriptor editWeddingDescriptor) {
+        assert weddingToEdit != null;
```
Comment (tingxuanp) [inline src/main/java/seedu/address/logic/commands/wedding/EditWeddingCommand.java:96]: Great that you used assertions here

File: src/test/java/seedu/address/logic/commands/wedding/EditWeddingCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.wedding.Wedding;
+import seedu.address.testutil.EditWeddingDescriptorBuilder;
+import seedu.address.testutil.WeddingBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EditCommand.
```
Comment (tingxuanp) [inline src/test/java/seedu/address/logic/commands/wedding/EditWeddingCommandTest.java:33]: minor nit/ typo? but EditWeddingCommand

File: src/test/java/seedu/address/logic/commands/wedding/ListWeddingsCommandIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for ListCommand.
```
Comment (tingxuanp) [inline src/test/java/seedu/address/logic/commands/wedding/ListWeddingsCommandIntegrationTest.java:15]: similarly ListWeddingCommand

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Person;
+import seedu.address.model.wedding.Wedding;
+import seedu.address.model.wedding.WeddingName;
+
+
+
+/**
+ * A utility class to help with building Person objects.
```
Comment (tingxuanp) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:13]: typo, Wedding

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    private WeddingName weddingName;
+    private int peopleCount;
+    private Person partner1;
+    private Person partner2;
+    private ArrayList<Person> guestList;
+    private Address address;
+    private String date;
+
+    /**
+     * Creates a {@code PersonBuilder} with the default details.
```
Comment (tingxuanp) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:37]: typo, WeddingBuilder

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+     */
+    public WeddingBuilder() {
+        weddingName = new WeddingName(DEFAULT_WEDDING_NAME);
+        //partner1 = new PersonBuilder().withName(DEFAULT_PARTNER1).build();
+        //partner2 = new PersonBuilder().withName(DEFAULT_PARTNER2).build();
```
Comment (tingxuanp) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:42]: should this be left commented out?

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        address = new Address(DEFAULT_ADDRESS);
+        date = DEFAULT_DATE;
+    }
+
+    /**
+     * Initializes the PersonBuilder with the data of {@code personToCopy}.
```
Comment (tingxuanp) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:48]: typo, `WeddingBuilder` and `weddingToCopy`

**Review by tingxuanp — COMMENTED** [review]
LGTM overall, couldnt find major issues just some typos to fix

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * Sets the {@code partner2} of the {@code Wedding} that we are building.
+     */
+    public WeddingBuilder withPartner2(Person partner2) {
+        this.partner2 = partner2;
+        return this;
+    }
+
+    /**
+     * Sets the {@code Address} of the {@code Person} that we are building.
```
Comment (DanzaSeah) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:76]: should it be Wedding we are building

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+     * Creates a {@code WeddingBuilder} with the default details.
+     */
+    public WeddingBuilder() {
+        weddingName = new WeddingName(DEFAULT_WEDDING_NAME);
+        address = new Address(DEFAULT_ADDRESS);
+        date = DEFAULT_DATE;
+    }
+
+    /**
+     * Initializes the WeddingBuilder with the data of {@code personToCopy}.
```
Comment (DanzaSeah) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:41]: Might have a typo should it be weddingToCopy instead

File: src/test/java/seedu/address/testutil/WeddingBuilder.java [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * Sets the {@code Address} of the {@code Person} that we are building.
+     */
+    public WeddingBuilder withAddress(String address) {
+        this.address = new Address(address);
+        return this;
+    }
+
+    /**
+     * Sets the {@code Date} of the {@code Person} that we are building.
```
Comment (DanzaSeah) [inline src/test/java/seedu/address/testutil/WeddingBuilder.java:84]: Should it be wedding instead of person

**Review by DanzaSeah — APPROVED** [review]
LGTM, just some typos

**Review by DanzaSeah — APPROVED** [review]
LGTM

**Pull request merged**